V2.2.1 top level build

RomainNaour opened the pull request:
Hello,

Here is another series to add a top level CMakeLists.txt which can be used with an automated build system (like Buildroot) to build all openpowerlink components.

It take into account Josef Baumgar's  comments which want to keep the ability to build separately each openpowerlink components. see [1]

Best regards,
Romain

[1] https://sourceforge.net/p/openpowerlink/discussion/newbie/thread/3f13af65/

RomainNaour commented:
Hello Josef,

What the build log says ?

Maybe we should drop the changes in  hardware/* ? I only use the part in drivers/linux/ 
Also, I can't test Altera and Xilink build right now.

Thanks,
Romain

jbaumgar commented:
Hello Romain,

currently we haven't time to examine the problem. I also would suggest to drop the changes in hardware.

Best regards,

Josef

```
[openPOWERLINK_2.3_Xilinx] $ /bin/bash /tmp/hudson450709699510803347.sh
INFO: Start reduced build to check SW only.
~/workspace/openPOWERLINK_2.3_Xilinx/hardware/build/xilinx-microblaze ~/workspace/openPOWERLINK_2.3_Xilinx
-- Found hardware platform: DEMO_S6PLKEB_MN_SINGLE_HOSTIF_DRV set to OFF!
-- Found hardware platform: DEMO_S6PLKEB_MN_SINGLE_HOSTIF_GPIO set to OFF!
-- Found hardware platform: DEMO_S6PLKEB_CN_SINGLE_GPIO set to OFF!
-- Found hardware platform: DEMO_LX150T_MN_DUAL_HOSTIF_GPIO set to OFF!
-- Found hardware platform: DEMO_Z702_MN_DUAL_SHMEM_GPIO set to OFF!
-- Configuring done
-- Generating done
-- Build files have been written to: /home/jenkins/workspace/openPOWERLINK_2.3_Xilinx/hardware/build/xilinx-microblaze
~/workspace/openPOWERLINK_2.3_Xilinx
~/workspace/openPOWERLINK_2.3_Xilinx/hardware/build/xilinx-microblaze ~/workspace/openPOWERLINK_2.3_Xilinx
CALL: cmake -DCMAKE_BUILD_TYPE=Debug -DDEMO_S6PLKEB_CN_SINGLE_GPIO=ON -DDEMO_S6PLKEB_MN_SINGLE_HOSTIF_DRV=ON -DDEMO_S6PLKEB_MN_SINGLE_HOSTIF_GPIO=ON -DDEMO_LX150T_MN_DUAL_HOSTIF_GPIO=ON -DDEMO_Z702_MN_DUAL_SHMEM_GPIO=ON -DSKIP_BITSTREAM=ON /home/jenkins/workspace/openPOWERLINK_2.3_Xilinx/hardware
-- Hardware platform: DEMO_S6PLKEB_MN_SINGLE_HOSTIF_DRV set to ON!
-- Generating targets for demo s6plkeb-mn-single-hostif-drv!
-- The C compiler identification is GNU 4.6.4
-- Check for working C compiler: /opt/toolchains/xilinx/14.7/ISE_DS/EDK/gnu/microblaze/lin/bin/mb-gcc
-- Check for working C compiler: /opt/toolchains/xilinx/14.7/ISE_DS/EDK/gnu/microblaze/lin/bin/mb-gcc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Hardware platform: DEMO_S6PLKEB_MN_SINGLE_HOSTIF_GPIO set to ON!
-- Generating targets for demo s6plkeb-mn-single-hostif-gpio!
-- Hardware platform: DEMO_S6PLKEB_CN_SINGLE_GPIO set to ON!
-- Generating targets for demo s6plkeb-cn-single-gpio!
-- Hardware platform: DEMO_LX150T_MN_DUAL_HOSTIF_GPIO set to ON!
-- Generating targets for demo lx150t-mn-dual-hostif-gpio!
-- Hardware platform: DEMO_Z702_MN_DUAL_SHMEM_GPIO set to ON!
-- Generating targets for demo z702-mn-dual-shmem-gpio!
-- Configuring done
CMake Error at drivers/openmac/CMakeLists.txt:93 (ADD_LIBRARY):
  Cannot find source file:

    /home/jenkins/workspace/openPOWERLINK_2.3_Xilinx/hardware/drivers/openmac/boards/avnet-s6plkeb/common/drivers/openmac/omethlib_phycfg.c

  Tried extensions .c .C .c++ .cc .cpp .cxx .m .M .mm .h .hh .h++ .hm .hpp
  .hxx .in .txx


CMake Error at drivers/openmac/CMakeLists.txt:93 (ADD_LIBRARY):
  Cannot find source file:

    /home/jenkins/workspace/openPOWERLINK_2.3_Xilinx/hardware/drivers/openmac/boards/avnet-s6plkeb/common/drivers/openmac/omethlib_phycfg.c

  Tried extensions .c .C .c++ .cc .cpp .cxx .m .M .mm .h .hh .h++ .hm .hpp
  .hxx .in .txx


CMake Error at drivers/openmac/CMakeLists.txt:93 (ADD_LIBRARY):
  Cannot find source file:

    /home/jenkins/workspace/openPOWERLINK_2.3_Xilinx/hardware/drivers/openmac/boards/avnet-lx150t/common/drivers/openmac/omethlib_phycfg.c

  Tried extensions .c .C .c++ .cc .cpp .cxx .m .M .mm .h .hh .h++ .hm .hpp
  .hxx .in .txx


CMake Error at drivers/openmac/CMakeLists.txt:93 (ADD_LIBRARY):
  Cannot find source file:

    /home/jenkins/workspace/openPOWERLINK_2.3_Xilinx/hardware/drivers/openmac/boards/xilinx-z702/common/drivers/openmac/omethlib_phycfg.c

  Tried extensions .c .C .c++ .cc .cpp .cxx .m .M .mm .h .hh .h++ .hm .hpp
  .hxx .in .txx


CMake Error: Cannot determine link language for target "omethlib-s6plkeb_mn_single_hostif_drv".
CMake Error: CMake can not determine linker language for target: omethlib-s6plkeb_mn_single_hostif_drv
CMake Error: Cannot determine link language for target "omethlib-s6plkeb_cn_single_gpio".
CMake Error: CMake can not determine linker language for target: omethlib-s6plkeb_cn_single_gpio
CMake Error: Cannot determine link language for target "omethlib-lx150t_mn_dual_hostif_gpio".
CMake Error: CMake can not determine linker language for target: omethlib-lx150t_mn_dual_hostif_gpio
CMake Error: Cannot determine link language for target "omethlib-z702_mn_dual_shmem_gpio".
CMake Error: CMake can not determine linker language for target: omethlib-z702_mn_dual_shmem_gpio
-- Generating done
-- Build files have been written to: /home/jenkins/workspace/openPOWERLINK_2.3_Xilinx/hardware/build/xilinx-microblaze
Build step 'Execute shell' marked build as failure
Archiving artifacts
[WARNINGS] Skipping publisher since build result is FAILURE

Deleting project workspace... 
Skipped based on build state FAILURE

Finished: FAILURE
```

RomainNaour commented:
Hello Josef,

I updated this series and remove hardware part.
Also, I removed unittest part since It not used in top level CMakeLists.txt.
I hope it'll fixes your issue.

Best regards,
Romain

jbaumgar commented:
The first two changes are integrated. We are not adding the top level CMakeLists.txt.
